imap_processing/idex/idex_l2a.py

```diff
@@ -215,3 +234,42 @@ def calculate_kappa(mass_scales: np.ndarray, peaks_2d: list) -> NDArray:
         ]
     )
     return kappas
+
+
+def calculate_snr(tof_high: xr.DataArray, hs_time: xr.DataArray) -> NDArray:
+    """
+    Calculate the signal-to-noise ratio.
+
+    Parameters
+    ----------
+    tof_high : xarray.DataArray
+        The time of flight array.
+    hs_time : xarray.DataArray
+        The high sampling time array.
+
+    Returns
+    -------
+    numpy.ndarray
+        Signal-to-noise ratio at each event.
+    """
+    # Find indices where Time (High Sampling) is between -7 and -5 ns (no signal yet)
+    # To determine the baseline noise
+    baseline_noise = np.where(
+        np.logical_and(
+            hs_time >= BaselineNoiseTime.START, hs_time <= BaselineNoiseTime.STOP
+        ),
+        tof_high.data,
```

Comment:
Do you need the `.data` in all of these calls? I thought xarray might be able to be coerced to numpy so you don't have to explicitly access the data attribute.